### PR TITLE
add GOVERNANCE.md

### PR DESCRIPTION
**Pull request by haircommander**
#### What type of PR is this?

> /kind api-change
> /kind bug
> /kind cleanup
> /kind dependency-change
> /kind deprecation
> /kind design

/kind documentation

> /kind failing-test
> /kind feature
> /kind flake

#### What this PR does / why we need it:
adds GOVERNANCE.md
#### Does this PR introduce a user-facing change?

```release-note
none
```

### CODE REVIEW

**Comment by haircommander** [comment]
```
ASK [clone kubernetes source repo] ********************************************
task path: /go/src/github.com/cri-o/cri-o/contrib/test/integration/build/kubernetes.yml:3
fatal: [localhost]: FAILED! => {"changed": false, "cmd": "/usr/bin/git ls-remote origin -h refs/heads/master", "msg": "fatal: unable to access 'https://github.com/kubernetes/kubernetes.git/': transfer closed with outstanding read data remaining", "rc": 128, "stderr": "fatal: unable to access 'https://github.com/kubernetes/kubernetes.git/': transfer closed with outstanding read data remaining\n", "stderr_lines": ["fatal: unable to access 'https://github.com/kubernetes/kubernetes.git/': transfer closed with outstanding read data remaining"], "stdout": "", "stdout_lines": []}
```
:cry: 
/retest

**Comment by fidencio** [comment]
As someone who has been around but not spending as much as I want in the project, I simply love this initiative!

File: GOVERNANCE.md [hunk 1]
```diff
@@ -37,7 +37,19 @@
...
+
+They may also opt to be come an [emeritus maintainer](#emeritus-maintainer)
+
+# Emeritus Maintainer
+
+An emeritus maintainer is a maintainer who used to spend a significant amount of time on the project, but now has refocused their efforts.
+
+The CRI-O community has the emeritus maintainer tier to recognize the impact and input early maintainers have had on the project, while also being realistic about their current committment level.
+
+There is no formal process for becoming an emeritus maintainer: a maintainer simply slows down on their contributions but does not remove themselves from the [OWNERS file](OWNERS.md).
```
Comment (fidencio) [inline GOVERNANCE.md:52]: @haircommander, I really like the "Emeritus" idea, GNOME dos something quite similar.

If I may make a suggestion, I'd prefer having the Emeritus Maintainers listed as so. By doing this new contributors will know who they can ping, who they can ask for help, etc.  With no clear distinction between both may be a slightly confusing for new contributors (I must admit it was slightly confusing for me when I first arrived here a few months ago).

File: GOVERNANCE.md [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+* Triage GitHub issues and perform pull request reviews for other maintainers and the community. The areas of specialization listed in [OWNERS.md](OWNERS.md) can be used to help with routing an issue/question to the right person.
+* Triage CI issues - file issues for known flaky CI runs or bugs, and either fix or find someone to fix any breakages.
+* During GitHub issue triage, apply all applicable [labels](https://github.com/cri-o/cri-o/labels) to each new issue. Labels are extremely useful for future issue follow up. Which labels to apply is somewhat subjective so just use your best judgment. A few of the most important labels that are not self explanatory are:
+  * **good first issue**: Mark any issue that can reasonably be accomplished by a new contributor with this label.
+  * **help wanted**: Unless it is immediately obvious that someone is going to work on an issue (and if so assign it), mark it help wanted.
+  * **CRI change**: If sufficiently fixing the issue involves a change to the CRI, this label is given. Often, maintainers won't have time to make such a change, but the change could be seen as good and as such the issue isn't closed.
```
Comment (TomSweeneyRedHat) [inline GOVERNANCE.md:25]: Do you have a 'kind/feature' label and do you want to also highlight that?

File: GOVERNANCE.md [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+There is no formal process for becoming an emeritus maintainer: a maintainer simply slows down on their contributions but does not remove themselves from the [OWNERS file](OWNERS.md).
+
+# Feature addition policy
+
+Features are added by simply adding a PR. The maintainers handle this the same as a bug fix PR, discussing whether the feature will be added in the PR page.
+
+# Conflict resolution and voting
+
+In general, we prefer that technical issues and maintainer membership are amicably worked out between the persons involved. If a dispute cannot be decided independently, the maintainers can be called in to decide an issue. If the maintainers themselves cannot decide an issue, the issue will be resolved by voting. The voting process is a simple majority in which each maintainer receives one vote. Votes not received within the span of one week are treated as an abstention.
```
Comment (TomSweeneyRedHat) [inline GOVERNANCE.md:60]: Any tie breaker procedures?  If you list them, you'll never need them, if you don't....
Perhaps just "In the case of a tie vote, the issue will be considered to have not passed."

File: GOVERNANCE.md [hunk 4]
```diff
@@ -0,0 +1,68 @@
+# CRI-O contributor tiers
```
Comment (saschagrunert) [inline GOVERNANCE.md:1]: Let's link the file from the README.md with a short sentence describing its intention.

Comment (TomSweeneyRedHat) [reply GOVERNANCE.md:1]: Good thought @saschagrunert I'd also suggest a link and a bit of explanatory verbiage on CONTRIBUTING.md

File: GOVERNANCE.md (hunk 2 above)
Comment (haircommander) [reply GOVERNANCE.md:25]: that's more for PRs, for issues we have `area/vm` but I don't think it's common enough to warrant adding right now

**Comment by haircommander** [comment]
a year and a quarter later, I've addressed the comments, and updated with some feedback from today's community meeting :smiley: 
PTAL @cri-o/cri-o-maintainers

File: GOVERNANCE.md [hunk 5]
```diff
@@ -0,0 +1,99 @@
+# CRI-O GOVERNANCE.md
+
+CRI-O, being a project whose purpose is linked to fulfilling the needs of Kubernetes, has a governance structure that mirrors many SIGs and sub-projects of Kubernetes. This document highlights important aspects of, and serves as a summary of the [Kubernetes' community documentation](https://www.kubernetes.dev/docs.
```
Comment (saschagrunert) [inline GOVERNANCE.md:3]:
```suggestion
CRI-O, being a project whose purpose is linked to fulfilling the needs of Kubernetes, has a governance structure that mirrors many SIGs and sub-projects of Kubernetes. This document highlights important aspects of, and serves as a summary of the [Kubernetes' community documentation](https://www.kubernetes.dev/docs).
```

File: GOVERNANCE.md [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+
+* If you'd like to have a more active role in the community, and one day hope to become an approver, you may become a reviewer to build trust and deepen your knowledge of the code base. 
+* To become a reviewer, an you may file an issue with [this template](https://github.com/cri-o/cri-o/issues/new?assignees=&labels=area%2Fgithub-membership&template=membership.yml&title=REQUEST%3A+New+membership+for+%3Cyour-GH-handle%3E).
+* To find a sponsor, please feel free to reach out to the approvers. The approver team is happy to foster growth within the CRI-O community.
+* 
+### Reviewer responsibilities
+
+* Once you are a reviewer, the CRI-O community will expect you to start contributing PRs, taking issues and doing code reviews under the guidance of current approvers.
+
+* Monitor our channel (#crio) on the Kubernetes Slack instance and answer questions where possible (delayed response is perfectly acceptable).
```
Comment (saschagrunert) [inline GOVERNANCE.md:29]: Let's add a link to the slack channel

File: GOVERNANCE.md [hunk 7]
```diff
@@ -0,0 +1,99 @@
...
+## Approver
+### How to become an approver
+
+* The barrier to become an approver is naturally higher than that of becoming a reviewer
+* There isn't an explicitly spelled out set of criteria that is required, but generally being an approver means you will be spending significant time maintaining and reviewing CRI-O code. 
+* To become an approver, a simple majority vote among the current approvers should be performed. See [conflict resolution](#conflict-resolution-and-voting) for more information on how to handle disputes.
+
+### Approver responsibilities
+
+* All of the responsibilities of the reviewers, and...
```
Comment (saschagrunert) [inline GOVERNANCE.md:47]:
```suggestion
Approvers have all of the responsibilities of the reviewers, as well as:
```

**Review by saschagrunert — COMMENTED** [review]
Awesome, just a few nits otherwise LGTM

File: .github/ISSUE_TEMPLATE/membership.yml [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+  attributes:
+    label: Requirements
+    options:
+    - label: I have reviewed the [community membership guidelines](https://git.k8s.io/community/community-membership.md)
+      required: true
+    - label: I have [enabled 2FA on my GitHub account](https://github.com/settings/security)
+      required: true
+    - label: I have one sponsor from the list of approvers in at least one OWNERS file within one of the CRI-O repository
+      required: true
+    - label: I have spoken to my sponsor ahead of this application, and they have agreed to sponsor my application
```
Comment (fgiudici) [inline .github/ISSUE_TEMPLATE/membership.yml:24]: Nit:
```suggestion
    - label: I have spoken to my sponsor ahead of this application, and he has agreed to sponsor my application
```

File: GOVERNANCE.md [hunk 9]
```diff
@@ -0,0 +1,99 @@
...
+## Community Member
+
+* Anyone can and is encouraged to be a community member! The CRI-O community eagerly accept PRs, issues, new users and questions.
+
+## Reviewer
+
+### How to become a reviewer
+
+* If you'd like to have a more active role in the community, and one day hope to become an approver, you may become a reviewer to build trust and deepen your knowledge of the code base. 
+* To become a reviewer, an you may file an issue with [this template](https://github.com/cri-o/cri-o/issues/new?assignees=&labels=area%2Fgithub-membership&template=membership.yml&title=REQUEST%3A+New+membership+for+%3Cyour-GH-handle%3E).
```
Comment (fgiudici) [inline GOVERNANCE.md:22]:
```suggestion
* To become a reviewer, you may file an issue with [this template](https://github.com/cri-o/cri-o/issues/new?assignees=&labels=area%2Fgithub-membership&template=membership.yml&title=REQUEST%3A+New+membership+for+%3Cyour-GH-handle%3E).
```

File: .github/ISSUE_TEMPLATE/membership.yml (hunk 8 above)
Comment (haircommander) [reply .github/ISSUE_TEMPLATE/membership.yml:24]: please my friend, allow me to introduce you to the [singular they](https://en.wikipedia.org/wiki/Singular_they), a gender neutral alternative to gendered pronouns. It's become the idiomatic way to have documentation be non-gendered and more inclusive :slightly_smiling_face:

File: GOVERNANCE.md (hunk 5 above)
Comment (haircommander) [reply GOVERNANCE.md:3]: good catch!

File: .github/ISSUE_TEMPLATE/membership.yml (hunk 8 above)
Comment (fgiudici) [reply .github/ISSUE_TEMPLATE/membership.yml:24]: ups, shame upon me 😅!

Comment (haircommander) [reply .github/ISSUE_TEMPLATE/membership.yml:24]: no shame, only a teachable moment :smile:

**Comment by haircommander** [comment]
updated!

**Review by fgiudici — COMMENTED** [review]
LGTM, great job!

**Comment by fidencio** [comment]
@haircommander, the document is in a very good shape.  One thing that I am still missing, though, is information on how to find the list of reviewers / approvers.  And, yes, I know,  such a list doesn't exist right now.

So, on this, where do you plan to have such a list?  I am thinking mostly on the scenario where I have interest in one area and I want to find folks from the community with knowledge in that area to give me some guidance.

**Comment by haircommander** [comment]
ah I see, I was planning on having https://github.com/cri-o/cri-o/blob/main/OWNERS be the authoritative document

**Comment by haircommander** [comment]
updated!

**Comment by fidencio** [comment]
LGTM, thanks @haircommander!

**Comment by fidencio** [comment]
@haircommander, one more thing, actually, the document doesn't cover the transition between reviewer -> contributor, maintainer -> reviewer or maintainer -> emeritus maintainer.

I think that would be nice to have documented as it will help to evaluate the situation of all the current members prsent in the https://github.com/cri-o/cri-o/blob/main/OWNERS file.

**Comment by haircommander** [comment]
> 

don't the sections "how to become x" cover that?

**Comment by fidencio** [comment]
> don't the sections "how to become x" cover that?

Kind of.  I'm thinking on the discussion we had Yesterday on the meeting.
Let's say, "12 months with zero contributions" revokes your status as approver / reviewer.  approvers are automatically moved to the emeritus status, for instance; reviewers get back to the contributor status.

Do you think it'd make sense to mention / make that explicit?

**Comment by haircommander** [comment]
okay, I've added a demotion section.

**Comment by fidencio** [comment]
Thanks, @haircommander!

I think it covers everything I could think of.

**Review by saschagrunert — APPROVED** [review]
/lgtm

**Pull request merged**